src/graph/interface/logical_tensor.cpp

```diff
@@ -194,10 +197,15 @@ status_t DNNL_API dnnl_graph_logical_tensor_init(
     val.layout_type = ltype;
     val.property = ptype;
 
-    // initialize the dims and strides
-    std::fill(val.dims, val.dims + DNNL_MAX_NDIMS, DNNL_GRAPH_UNKNOWN_DIM);
-    std::fill(val.layout.strides, val.layout.strides + DNNL_MAX_NDIMS,
-            DNNL_GRAPH_UNKNOWN_DIM);
+    if (ndims == 0) {
```

Comment:
why the change?

Reply:
We need to make sure that the uninitialized fields will not affect the hash results.